Neos.Flow/Classes/Http/Middleware/MethodOverrideMiddleware.php

```diff
@@ -0,0 +1,39 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Flow\Http\Middleware;
+
+use Neos\Flow\Annotations as Flow;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+/**
+ * Allows to override the request HTTP Method via different overrides in this order:
+ * - a "__method" argument passes in via request body
+ * - X-Http-Method-Override header
+ * - X-Http-Method header
+ */
+class MethodOverrideMiddleware implements MiddlewareInterface
+{
+    /**
+     * Process an incoming server request and overwrite the request method of a POST based on a __method argument or either
+     * the X-Http-Method-Override or X-Http-Method header.
+     */
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $next): ResponseInterface
+    {
+        if ($request->getMethod() === 'POST') {
+            $arguments = $request->getParsedBody();
```

Comment:
Good catch, it's filled with $_POST but not with eg. parsed JSON, for now I would argue that is fine, because would you want to bury the __method in a JSON request for an API? Seems tricky.

Reply:
☝️ `$_POST` (https://github.com/guzzle/psr7/blob/master/src/ServerRequest.php#L179) so for POST requests, which we accept only, it contains the parsed form/multipart parameters.
If we did want to parse any other content types, like for JSON APIs, we'd need to juggle the order of `parseBody` to come *before* routing and `overrideMethod`. Doesn't feel right and isn't how it worked before either. We could document this though or add it as a feature in an upcoming release if it's seen useful